### DISCUSSION TITLE
Is quic client can select a specific send port  to connect with quic server?

### DISCUSSION THREAD

**Discussion by Ang0426** [microsoft/msquic#3473] [Q&A]
Is quic client can select a specific send port  to connect with quic server and send data?
How to do it ?   Thanks

**Comment by nibanks** [comment]
```c
//
// Uses the QUIC (client) handle to start a connection attempt to the
// remote server. Can be passed either a connection or stream handle.
//
typedef
_IRQL_requires_max_(DISPATCH_LEVEL)
QUIC_STATUS
(QUIC_API * QUIC_CONNECTION_START_FN)(
    _In_ _Pre_defensive_ HQUIC Connection,
    _In_ _Pre_defensive_ HQUIC Configuration,
    _In_ QUIC_ADDRESS_FAMILY Family,
    _In_reads_or_z_opt_(QUIC_MAX_SNI_LENGTH)
        const char* ServerName,
    _In_ uint16_t ServerPort // Host byte order
    );
```
When you call `ConnectionStart` you specify the `ServerPort`. So yes, you can pick the destination port to use.

If you're talking about the **local** port, you can `SetParam` for `QUIC_PARAM_CONN_LOCAL_ADDRESS` to set a local port before calling start.

**Comment by Ang0426** [comment]
Thank you very much, I mean the local port , I will try QUIC_PARAM_CONN_LOCAL_ADDRESS
